### PR TITLE
[Addon]canary deploy step to do rollout in workflow

### PR DESCRIPTION
**Pull request by wangyikewxgm**
### Checklist

I have:

- [x] Title of the PR starts with type (e.g. `[Addon]` , `[example]` or `[Doc]`).
- [ ] Updated/Added any relevant [documentation](https://kubevela.io/docs/reference/addons/overview) and [examples](https://github.com/kubevela/catalog/tree/master/examples).
- [ ] New addon should be put in [experimental](https://github.com/kubevela/catalog/tree/master/experimental/addons).
- [x] Update addon should modify the `version` in `metadata.yaml` to generate a new version.

####  Verified Addon promotion rules

If this pr wants to promote an experimental addon to verified, you must check whether meet these conditions too:
  - [ ] This addon must be tested by addon's [e2e-test](./test/e2e-test/addon-test) to guarantee this addon can be enabled successfully.
  - This addon must have some basic but necessary information.
    - [ ] An accessible icon url and source url defined in addon's `metadata.yaml`.
    - [ ] A detail introduction include a basic example about how to use and what's the benefit of this addon in `README.md`.
    - [ ] Also provide an introduction in KubeVela [documentation](https://kubevela.net/docs/reference/addons/overview).
    - [ ] It's more likely to be accepted if useful examples provided in example [dir](examples/).

### CODE REVIEW

File: addons/kruise-rollout/definitions/canary-deploy.cue [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+					}]
+				}]
+			}
+		}]
+	}
+	parameter: {
+		//+usage=Declare the policies that used for this deployment. If not specified, the components will be deployed to the hub cluster.
+		policies?: *[] | [...string]
+		//+usage=Specify which component(s) to use for the canary rollout. If you do not specify, all components will be affected.
+		component?: string
```
Comment (barnettZQG) [inline addons/kruise-rollout/definitions/canary-deploy.cue:51]:
```suggestion
		component?: *[] | [...string]
```

Comment (wangyikewxgm) [reply addons/kruise-rollout/definitions/canary-deploy.cue]: fix

File: examples/kruise/rollout/rollout-with-alb.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        properties:
+          component: canary-demo
+          weight: 50
+          trafficRoutingType: "aliyun-alb"
+      - name: suspend-again
+        type: suspend
+      - type: canary-deploy
+        name: rollout-100
+        properties:
+          component: canary-demo
```
Comment (barnettZQG) [inline examples/kruise/rollout/rollout-with-alb.yaml:45]:
```suggestion
          components: ["canary-demo"]
```

**Review by barnettZQG — APPROVED** [review]
lgtm

**Review by barnettZQG — APPROVED** [review]
lgtm

**Pull request merged**